src/librustc_typeck/check/pat.rs

```diff
@@ -65,6 +65,8 @@ struct TopInfo<'tcx> {
     ///              found type `std::result::Result<_, _>`
     /// ```
     span: Option<Span>,
+    // This refers to the parent pattern, if any, in the case of a struct field pattern.
```

Comment:
You changed things so it doesn't just apply to struct patterns, right? So this comment is outdated.